workbench-core/accounts/src/services/projectService.ts

```diff
@@ -80,4 +71,206 @@ export default class ProjectService {
+    // Verify caller is an IT Admin--TODO implement after dynamic AuthZ
+    // const userGroupsForCurrentUser: string[] = await this._dynamicAuthorizationService.getUserGroups(user.id);
+    // if(userGroupsForCurrentUser.length !== 1 || userGroupsForCurrentUser[0] !== 'ITAdmin') {
+    //   throw Boom.forbidden('Only IT Admins are allowed to create new Projects.');
+    // }
+
+    // Verify project name is unique and cost center exists
+    const resultsFromValidityChecks = await Promise.all([
+      this._isProjectNameInUse(params.name),
+      this._getCostCenter(params.costCenterId)
+    ]);
+    const costCenter: CostCenter = resultsFromValidityChecks[1];
+
+    // Generate Project ID
+    const projectId = uuidWithLowercasePrefix(resourceTypeToKey.project);
+
+    // Create ProjectAdmin and Researcher groups--TODO implement after dynamic AuthZ
+    // const createProjectAdminGroupResponse = this._dynamicAuthorizationService.createGroup(user, {
+    //   groupId: `${projId}#PA`,
+    //   description: `Project Admin group for ${projId}`
+    // });
+    // const createResearcherGroupResponse = this._dynamicAuthorizationService.createGroup(user, {
+    //   groupId: `${projId}#Researcher`,
+    //   description: `Researcher group for ${projId}`
+    // });
+    // if (!createProjectAdminGroupResponse.created || !createResearcherGroupResponse.created) {
+    //   throw Boom.badImplementation(
+    //     'Failed to create Project Admin group or Researcher group with dynamic authorization service.'
+    //   );
+    // }
+
+    // Create Permissions for the groups--TODO implement after dynamic AuthZ
+    // const identityPermissions: IdentityPermission[] = this._createIdentityPermissionsForProject(projectId);
+    // const createIdentityPermissionsResponse = this._dynamicAuthorizationService.createIdentityPermissions(
+    //   user,
+    //   identityPermissions
+    // );
+    // if (!createIdentityPermissionsResponse.created) {
+    //   throw Boom.badImplementation(
+    //     'Failed to create batch identity permissions for project with dyamic authorization service.'
+    //   );
+    // }
+
+    // Create Proj in DDB
+    const newProject: Project = {
+      id: projectId,
+      createdAt: new Date().toISOString(),
+      costCenterId: params.costCenterId,
+      description: params.description,
+      name: params.name,
+      updatedAt: new Date().toISOString(),
+      status: 'AVAILABLE',
+      // Acc Metadata (get from cost center)
+      subnetId: costCenter.subnetId,
+      vpcId: costCenter.vpcId,
+      envMgmtRoleArn: costCenter.envMgmtRoleArn,
+      externalId: costCenter.externalId,
+      encryptionKeyArn: costCenter.encryptionKeyArn,
+      environmentInstanceFiles: costCenter.environmentInstanceFiles,
+      hostingAccountHandlerRoleArn: costCenter.hostingAccountHandlerRoleArn,
+      awsAccountId: costCenter.awsAccountId,
+      accountId: costCenter.accountId
+    };
+    try {
```

Comment:
In DDB, there should be a field for `dependency` that stores the `accountId`. However, the DDB entry should not include the `accountId` field. See [here](https://github.com/aws-solutions/solution-spark-on-aws/blob/develop/workbench-core/accounts/src/services/costCenterService.ts#L73-L81) for an example of how you can structure the code.

Reply:
Went ahead and wrote a method for this. Thanks for the code link.